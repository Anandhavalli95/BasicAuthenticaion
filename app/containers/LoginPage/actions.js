import {
  SET_LOGIN_DETAILS,
  GET_VALID_CREDS,
  GET_VALID_CREDS_FAILURE,
  GET_VALID_CREDS_SUCCESS,
} from './constants';

export function setLoginDetails(field, value) {
  return {
    type: SET_LOGIN_DETAILS,
    field,
    value,
  };
}

export function getValidCred() {
  return {
    type: GET_VALID_CREDS,
  };
}

export function getValidCredSuccess(details) {
  return {
    type: GET_VALID_CREDS_SUCCESS,
    details,
  };
}

export function getValidCredFailure(error) {
  return {
    type: GET_VALID_CREDS_FAILURE,
    error,
  };
}
