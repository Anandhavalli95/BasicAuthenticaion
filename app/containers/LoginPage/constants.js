export const SET_LOGIN_DETAILS = 'boilerplate/Login/SET_LOGIN_DETAILS';

export const GET_VALID_CREDS = 'boilerplate/Login/GET_VALID_CREDS';
export const GET_VALID_CREDS_SUCCESS =
  'boilerplate/Login/GET_VALID_CREDS_SUCCESS';
export const GET_VALID_CREDS_FAILURE =
  'boilerplate/Login/GET_VALID_CREDS_FAILURE';
