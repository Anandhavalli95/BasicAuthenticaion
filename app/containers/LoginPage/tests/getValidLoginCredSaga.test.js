import { put, takeLatest } from 'redux-saga/effects';

import { GET_VALID_CREDS } from '../constants';
import { getValidCredSuccess, getValidCredFailure } from '../actions';

import fetchValidCredWatcher, {
  fetchValidCredSaga,
} from '../getValidLoginCredentialsSaga';

describe('getValidCred Saga', () => {
  let getValidCredGenerator;

  beforeEach(() => {
    getValidCredGenerator = fetchValidCredSaga();

    const callDescriptor = getValidCredGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the getValidCredSuccess action if the response is received successfully', () => {
    const response = {
      email: 'admin@gmail.com',
      password: 'admin@123',
    };
    const putDescriptor = getValidCredGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(getValidCredSuccess(response)));
  });

  it('should call the getValidCredError action if the response errors', () => {
    const response = new Error('Something went wrong');
    const putDescriptor = getValidCredGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(getValidCredFailure(response)));
  });
});

describe('fetchProductList Watcher', () => {
  const fetchCredWatcher = fetchValidCredWatcher();

  it('should start task to watch for GET_VALID_CREDS action', () => {
    const takeLatestDescriptor = fetchCredWatcher.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(GET_VALID_CREDS, fetchValidCredSaga),
    );
  });
});
