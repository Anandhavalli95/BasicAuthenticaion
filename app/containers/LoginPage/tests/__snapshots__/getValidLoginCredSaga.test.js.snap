// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getValidCred Saga should call the getValidCredError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://68830be8-b6fb-4192-957d-e020f745ca90.mock.pstmn.io/getValidCred",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getValidCred Saga should dispatch the getValidCredSuccess action if the response is received successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://68830be8-b6fb-4192-957d-e020f745ca90.mock.pstmn.io/getValidCred",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
